DisplayWriterWorker.h

```diff
@@ -1,179 +1,201 @@
-        if (next_avail_pos != -1) {
-            game_board[next_avail_pos].is_current = true;
-        }
-
-        event_queue.submit_event(new CheckEndConditionEvent(player));
-    }
-
-public:
-    DisplayWriterWorker(GameEventQueue &eventQueue, DisplayWriter *displayWriter, std::vector<MatrixCell> &gameBoard)
-            : Worker(eventQueue), display_writer(displayWriter), game_board(gameBoard) {}
-
-    void handle_event(GameEvent *event) override {
-        switch (event->get_event_type()) {
-            case GameEventType::MOVE_PLAYER_PLACEHOLDER:
-                move_player_placeholder(dynamic_cast<MovePlayerPlaceholderEvent*>(event));
-                break;
-            case GameEventType::WRITE_PLAYER_SYMBOL:
-                write_player_symbol(dynamic_cast<WritePlayerSymbolEvent*>(event));
-                break;
-            case GameEventType::WRITE_MATRIX:
-                display_writer->write_grid();
-                break;
-            case GameEventType::WRITE_PLAYER_PLACEHOLDER: {
-                auto *write_placeholder_ev = dynamic_cast<WritePlayerPlaceholderEvent*>(event);
-                auto player = write_placeholder_ev->get_player();
-                auto coord = write_placeholder_ev->get_coordinate();
-
-                game_board[coord.y * 3 + coord.x].state = get_placeholder(player);
-                display_writer->write_placeholder_for(player.get_symbol(), coord);
-
-                event_queue.submit_event(new WaitPlayerInputEvent);
-                break;
-            }
-            case GameEventType::WRITE_STROKE: {
-                auto *write_stroke_ev = dynamic_cast<WriteStrokeEvent*>(event);
-                display_writer->write_stroke(write_stroke_ev->get_coord(), write_stroke_ev->get_direction());
-                break;
-            }
-            case GameEventType::WRITE_VICTORY_MSG: {
-                auto *write_victory_msg_ev = dynamic_cast<WriteVictoryMessageEvent*>(event);
-                display_writer->write_victory_msg_for(write_victory_msg_ev->get_player_id());
-                event_queue.submit_event(new ShutdownEvent);
-                break;
-            }
-            case GameEventType::WRITE_DRAW_MSG:
-                display_writer->write_draw_msg();
-                event_queue.submit_event(new ShutdownEvent);
-                break;
-        }
-    }
-
-    std::unordered_set<GameEventType> get_supported_event_types() const override {
-        return {GameEventType::WRITE_MATRIX, GameEventType::WRITE_PLAYER_PLACEHOLDER,
-                GameEventType::WRITE_PLAYER_SYMBOL, GameEventType::MOVE_PLAYER_PLACEHOLDER,
-                GameEventType::WRITE_STROKE, GameEventType::WRITE_DRAW_MSG, GameEventType::WRITE_VICTORY_MSG };
-    }
-
-};
-
-#endif //TICTACTOE_DISPLAYWRITERWORKER_H
+#ifndef TICTACTOE_DISPLAYWRITERWORKER_H
+#define TICTACTOE_DISPLAYWRITERWORKER_H
+
+#include <vector>
+#include "InvalidPath.h"
+#include "Worker.h"
+#include "Windows.h"

```

Comment:
Try not to include platform-specific headers without checking if it is supported